### PR TITLE
Provide `linux/arm64` Docker images

### LINKED ISSUE #78: Publish images for the `linux/arm64` Docker platform
In #76 I mentioned a request to publish `linux/arm64` Docker images. Currently, building on M1 is unbearably slow in many cases due to the virtualization overhead. I’ve almost given up on running RCD locally as a result.

Using `linux/arm64` images removes the virtualization overhead and makes the local development experience much better.

### PR DESCRIPTION
**Pull request by ianks**
When using a Macbook M1,`rake-compiler-dock` images have to run in emulation mode which makes things considerably slower. Docker also spits out a warning about it:

```
WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
```

This PR adds a step to the CI matrix which publishes `linux/arm64` versions to Docker hub, which will make my CPU much happier :)

resolves #78

### CODE REVIEW

**Comment by ianks** [comment]
Can someone trigger a re-run please? :)

**Comment by ianks** [comment]
Hmmm, not sure if this error is related to my change set or not, but both amd64 and arm64 are failing on autogen.sh

**Comment by flavorjones** [comment]
I've opened up #80 to investigate what's going on with the builds. Seems like bitrot.

**Comment by sodabrew** [comment]
Can rebase this PR following the #81 merge?

**Comment by flavorjones** [comment]
@ianks you may want to rebase again since I just merged https://github.com/rake-compiler/rake-compiler-dock/pull/82

**Comment by ianks** [comment]
Alright, here's a summary after I let the builds run overnight:

- ✅ x86-mingw32
- ❌ x86-mingw-ucrt - not finding the built-from-source arm64 version of binutils
- ✅ x64-mingw32
- ❌ x86_64-linux - choking on building Ruby coroutines on Ruby 3.0
- ✅ arm-linux
- ❌ aarch64-linux - `rb_thread_call_without_gvl() not found (RuntimeError)`
- ✅ jruby

I need to dive in more to debug these, but figured I'd post a summary in case someone can give a helpful pointer. Wouldn't mind rubber fucking this a bit!

**Comment by flavorjones** [comment]
Damn, I think my refactoring of `Dockerfile.mri.erb` is going to make you have to resolve some minor conflicts. Sorry for the inconvenience.

**Comment by flavorjones** [comment]
Also - I was planning on working on https://github.com/rake-compiler/rake-compiler-dock/issues/71 this weekend, which would publish "nightly" images to GHCR. If that meets your needs you may be able to drop the tarball upload step in this PR.

File: build/mk_osxcross.sh [hunk 1]
```diff
@@ -13,7 +13,7 @@ set -x
 curl -L -o MacOSX11.1.sdk.tar.xz https://github.com/larskanis/MacOSX-SDKs/releases/download/11.1/MacOSX11.1.sdk.tar.xz
 tar -xf MacOSX11.1.sdk.tar.xz -C .
 cp -rf /usr/lib/llvm-10/include/c++ MacOSX11.1.sdk/usr/include/c++
-cp -rf /usr/include/x86_64-linux-gnu/c++/9/bits/ MacOSX11.1.sdk/usr/include/c++/v1/bits
+cp -rf /usr/include"$(uname -m)"-linux-gnu/c++/9/bits/ MacOSX11.1.sdk/usr/include/c++/v1/bits
```
Comment (flavorjones) [inline build/mk_osxcross.sh:16]: should be `/usr/include/$(uname -m)-linux-gnu ...` (note the "/")

**Comment by ianks** [comment]
This branch is kind of FUBAR at the moment, and I'm having trouble piecing things together. TBH, I'm not sure about the feasibility of this patch with the current complexity of the Dockerfiles, etc

**Comment by flavorjones** [comment]
@ianks OK, don't feel pressured to land this in time for Ruby 3.2 support, we can add it in a 1.3.x patch release when it's ready.

**Pull request closed without merging**